app/unit-tests/src/org/commcare/android/util/TestUtils.java

```diff
@@ -97,21 +99,27 @@ public TransactionParser getParser(KXmlParser parser) {
                     return new FormInstanceXmlParser(parser, CommCareApplication.instance(),
                             Collections.unmodifiableMap(formInstanceNamespaces),
                             CommCareApplication.instance().getCurrentApp().fsPath(GlobalConstants.FILE_CC_FORMS));
-                } else if(CaseXmlParser.CASE_XML_NAMESPACE.equals(parser.getNamespace()) && "case".equalsIgnoreCase(parser.getName())) {
+                } else if (CaseXmlParser.CASE_XML_NAMESPACE.equals(parser.getNamespace()) && "case".equalsIgnoreCase(parser.getName())) {
 
                     //Note - this isn't even actually bulk processing. since this class is static
                     //there's no good lifecycle to manage the bulk processor in, but at least
                     //this will validate that the bulk processor works.
-                    if(bulkProcessingEnabled)  {
-                        return new AndroidBulkCaseXmlParser(parser, getCaseStorage(db), new EntityStorageCache("case", db), new AndroidCaseIndexTable(db)) {
+                    EntityStorageCache entityStorageCache;
+                    try {
+                        entityStorageCache = new EntityStorageCache("case", db, AppUtils.getCurrentAppId());
+                    } catch (Exception e) {
```

Comment:
If the only reason this ever occurs is the null currentAppId we should check for that explicitly, rather than catch a generic exception here, which will hide other errors that we aren't anticipating.

Reply:
Agree. Updated.